controllers/provider-aws/example/Infrastructure.yaml

```diff
@@ -0,0 +1,26 @@
+---
+apiVersion: extensions.gardener.cloud/v1alpha1
+kind: Infrastructure
+metadata:
+  name: infrastructure
+spec:
+  type: aws
+  providerConfig:
+    apiVersion: aws.provider.extensions.gardener.cloud/v1alpha1
+    kind: InfrastructureConfig
+    networks:
+      vpc:
+        cidr: 10.250.0.0/16
+      internal:
+        - 10.250.112.0/22
```

Comment:
I'm not happy how the Shoot specification turned out with respect to zones and multiple networks. It's not very intuitive.

I think that a more reasonable approach would be a something on the line of:

```yaml
apiVersion: extensions.gardener.cloud/v1alpha1
kind: Infrastructure
metadata:
  name: infrastructure
spec:
  type: aws
  providerConfig:
    apiVersion: aws.provider.extensions.gardener.cloud/v1alpha1
    kind: InfrastructureConfig
    networks:
      vpc:
        cidr: 10.250.0.0/16
      zones:
      - name: eu-west-1a
        internal:
        - 10.250.112.0/22
        public:
        - 10.250.96.0/22
        workers:
        - 10.250.0.0/19
  region: eu-west-1
  secretRef:
    name: cloudprovider
    namespace: shoot--dev--azaalouk-test-112
  sshPublicKey: c3NoLXJzYSBBQUFBQjNOemFDMXljMkVBQUFBREFRQUJBQUFDQVFEbk5rZkkxSWhBdGMyUXlrQ2sxTXNEMGpyNHQwUTR3OG9ZQkk0M215eElGc1hTRWFoQlhGSlBEeGl3akQ2KzQ1dHVHa0x2Y2d1WVZYcnFIOTl5eFM3eHpRUGZmdU5kelBhTWhIVjBHRFZIVDkyK2J5MTdtUDRVZDBFQTlVR29KeU1VeUVxZG45b1k1aURSUktRVHFzdW5QR0hpWVVnQ3ZPMElJT0kySTNtM0FIdlpWN2lhSVhKVE53eGE3ZVFTVTFjNVMzS2lseHhHTXJ5Y3hkNW83QWRtVTNqc3JhMVdqN2tjSFlseTVINkppVExsY0FxNVJQYzVXOUhnTHhlODZnUXNzN2pZN2t5NXJ1elBZV3ppdS94QlZBNGJQRXhVY2dIL3ZZTnl0aWg4OTBHWGRlcm1IOW5QSXpRZWlSWUlMdzJsaEMrdzBMdjM3QXdBYVNWRFlnY3NWNkdENllKaXN3VFV5ZStXdU9iZm1nWlFqaUppbUkwWWlrY2U2d3l2MFRHUW1BM3lnVDE1MDBoMnZMWXNMdWJJRjZGNkJRcTlKcDZ0M0w2RENoMmgvY3RSZEl2SXE2SWRPQnpOeGl4V2trbHJQbkhwS3B3eFEzVVJDRDRHMHhBK3dWZmtML05ueVhDSGM2Qk0zVUNhVDBpdExycjkwRGFTNWFvYVVGVHJuS2tDN1JxUWlwU3ZYVUcrQ1RqWnljLzRsblFOOSt6WmwvVE05QmxTYTQ3VGc1Myt6NjcxSmhRZXNBNUIrNVRtSFNGdHgwbXFzWnRJSng4dEtyR1VPeG1tTTVVb2J4VGp2TXBrMWpJWU4vWFJOdCt4R2VSbFVEZW9xalJMZnJOdjljZFF4Z0hzZXhmd3VUeERHYjlnb21RR0hRSjQrMW1kYjVUK2NmV0pUUTNCQXc9PQ==
```

Reply:
👍 thanks @mvladev, very good and valid suggestion.
I'd prefer this:

```yaml
apiVersion: aws.provider.extensions.gardener.cloud/v1alpha1
kind: InfrastructureConfig
networks:
  vpc:
    cidr: 10.200.0.0/16
  internal:
  - zone: eu-west-1a
    cidr: 10.250.112.0/22
  - zone: eu-west-1b
    cidr: 10.250.<whatever>
  public:
  - zone: eu-west-1a
    cidr: 10.250.96.0/22
  - zone: eu-west-1b
    cidr: 10.250.<whatever>
  workers:
  - zone: eu-west-1a
    cidr: 10.250.0.0/19
  - zone: eu-west-1b
    cidr: 10.250.<whatever>
```

(Make `internal`,`public`,`workers` "leading" below the `networks` field instead of `zones` "leading").

OK?

Reply:
@rfranzke I would argue for my approach, because this is the way networking is implemented in AWS:

- In 1 Region you have **1 + N** availability zones
- In 1 Region you have **1 + N** VPCs
- In one VPC you can have **1 + N** subnets each residing in only **1** availability zone.

Reply:
I got the point, but shouldn't our spec rather reflect what WE (with our clusters managed by gardener) support in our AWS implementation (in contrast to what is possible in AWS in general)?
For example, we only provide exactly one internal,public,worker subnet per zone for our AWS shoots, not `n`.
Hence, adapting your spec accordingly would result in
```yaml
apiVersion: aws.provider.extensions.gardener.cloud/v1alpha1
kind: InfrastructureConfig
networks:
  vpc:
    cidr: 10.200.0.0/16
  zones:
  - name: eu-west-1a
    internal: 10.250.112.0/22
    public: 10.250.96.0/22
    workers: 10.250.0.0/19
  - name: eu-west-1b
    internal: 10.220.112.0/22
    public: 10.220.96.0/22
    workers: 10.220.0.0/19
```

Though, I still favour `internal`/`public`/`workers` as direct child below `networks` instead of `zones` (it just "feels" more naturally). But I can also live with your approach. In the end it's important that it gets more declarative which CIDR is used in which zone.
@zanetworker @adracus @timuthy opinions? Let's just use the approach the majority of us likes most.

Reply:
@rfranzke 

> For example, we only provide exactly one internal,public,worker subnet per zone for our AWS shoots, not `n`.

this seems okay.

Reply:
I vote for:
```yaml
networks:
      vpc:
        cidr: 10.200.0.0/16
      zones:
      - name: eu-west-1a
        internal: 10.250.112.0/22
        public: 10.250.96.0/22
        workers: 10.250.0.0/19
      - name: eu-west-1b
        internal: 10.220.112.0/22
        public: 10.220.96.0/22
        workers: 10.220.0.0/19
```

Reply:
I'd vote for the solution that is easier to validate + more difficult to get wrong.
If for each zone, exactly one each network is required, then I guess I'd go with the second approach @rfranzke has posted (zone name + immediate cidr values).

Rationale:
* In @mvladev 's original approach, it doesn't reflect what we're actually doing
* In @rfranzke 's original approach, it'd be easy for me to forget putting in a cidr for one of the networks in a certain zone.

Reply:
Thanks, then let's go with it!